woocommerce/woocommerce
# Supply a fallback queryId if not provided by pattern.

Also eliminate stashing of `$this->parsed_block` as parsed block can be modified at `render_block_data` hook (meaning the block can be modified *after* it was stashed.

### Submission Review Guidelines:

* I have followed the [WooCommerce Contributing Guidelines](<https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md>) and the [WordPress Coding Standards](<https://make.wordpress.org/core/handbook/best-practices/coding-standards/>).
* I have checked to ensure there aren't other open [Pull Requests](<https://github.com/woocommerce/woocommerce/pulls>) for the same update/change.
* I have reviewed my code for [security best practices](<https://developer.wordpress.org/apis/security/>).
* Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes woocommerce/woocommerce#52457 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](<https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions>), include your detailed testing instructions:

Add snippet

```
function kia_insert_manual_collection() {
	$blocks = '
	<div class="wp-block-group alignwide" style="margin-top:0px;margin-bottom:80px;padding-top:0;padding-right:var(--wp--preset--spacing--30);padding-bottom:0;padding-left:var(--wp--preset--spacing--30)">

			
			<h2 class="wp-block-heading">Latest products</h2>
			

		
		<div class="wp-block-woocommerce-product-collection alignwide">
			
			

			

			

			
			
		</div>
		
	</div>
	
	';
	echo do_blocks($blocks);
}
add_action( 'woocommerce_simple_add_to_cart', 'kia_insert_manual_collection', 20 );
```

And go to single product page for a simple product. It should add a product collection showing the 3 latest products.

With Debug enabled, you should see a complaints about undefined array keys

```
Warning: Undefined array key “queryId” in /app/wp-content/plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php on line 601 
Warning: Undefined array key “queryId” in /app/wp-content/plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php on line 349 
Warning: Undefined array key “queryId” in /app/wp-content/plugins/woocommerce/src/Blocks/BlockTypes/ProductCollection.php on line 623 
```

With this PR, these notices should go away.

### Changelog entry


<details><summary>Changelog Entry Details</summary>

#### Significance


#### Type


#### Message

</details>

<details><summary>Changelog Entry Comment</summary>

#### Comment

</details>
